docs/guides/version-2-upgrade.md

```diff
@@ -0,0 +1,102 @@
+---
+page_title: "Terraform Coder Provider Version 2 Upgrade Guide"
+---
+
+# Terraform Coder Provider Version 2 Upgrade Guide
+
+Version 2.0.0 of the Coder provider for Terraform is a major release that removes and introduces some changes that you will need to consider when upgrading.
+This guide is intended to help with the process and focus only on the changes from version 1.X to version 2.0.0.
+
+
+!> Using Version 2.0.0 of the Coder provider requires Coder Server version [`2.13.0`](https://github.com/coder/coder/releases/tag/v2.13.0) or later.
+
+Upgrade topics:
+
+- [Provider Version Configuration](#provider-version-configuration)
+- [Provider Arguments](#provider-arguments)
+- [Data Source: coder_git_auth --> coder_external_auth](#data-source-coder_git_auth)
+- [Data Source: coder_workspace](#data-source-coder_workspace)
+
+## Provider Version Configuration
+
+-> Before upgrading to version 2.0.0, it is recommended to upgrade to the most recent 1.X version and ensure that your environment successfully runs [`terraform plan`](https://developer.hashicorp.com/terraform/cli/commands/plan) without unexpected changes or deprecation notices.
+
+We recommended using [version constraints](https://developer.hashicorp.com/terraform/language/providers/requirements#version-constraints) when configuring Terraform providers.
+
+
+For example, given the previous configuration:
+
+```terraform
+terraform {
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = "~> 1.0.0"
+    }
+  }
+}
+
+provider "coder" {
+  feature_use_managed_variables = true
+}
+```
+
+Update to the latest 2.X version:
+
+```terraform
+terraform {
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = "~> 2.0.0"
+    }
+  }
+}
+
+provider "coder" {}
+```
+
+## Provider Arguments
+
+Version 2.0.0 removes the [`feature_use_managed_variables`](https://registry.terraform.io/providers/coder/coder/1.0.4/docs#feature_use_managed_variables-1) argument from the `provider` block.
```

Comment:
Maybe we should explain that this has been deprecated for a long time and has no impact on Coder workspace anymore.